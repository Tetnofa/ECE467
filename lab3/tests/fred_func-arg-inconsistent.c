int wrong_return();

void wrong_return() {}

int wrong_arg_type(int a);

int wrong_arg_type(bool a) { return 0; }

int different_arg_names(int a, int b);

int different_arg_names(int b, int a) {
  return 0;
}

// Assume that there is at most 1 proto and at most 1 def for a given name
// aka, this is not something we need to handle.
// int redef_proto();
// int redef_proto();
// int overload(int a);
// int overload(bool b);
