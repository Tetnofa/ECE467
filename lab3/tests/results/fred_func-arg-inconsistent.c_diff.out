0a1,34
> Program [useIO=0](1,0)
> 	Function[isProto=1]	Int(1,0)
> 		Identifier[name:wrong_return](1,4)
> 	Function[isProto=0]	Void(3,0)
> 		Identifier[name:wrong_return](3,5)
> 		Scope(3,20)
> 	Function[isProto=1]	Int(5,0)
> 		Identifier[name:wrong_arg_type](5,4)
> 		Parameter		Int(5,19)
> 			Identifier[name:a](5,23)
> 	Function[isProto=0]	Int(7,0)
> 		Identifier[name:wrong_arg_type](7,4)
> 		Parameter		Bool(7,19)
> 			Identifier[name:a](7,24)
> 		Scope(7,27)
> 			ReturnStmt(7,29)
> 				Int Expression(7,36)
> 					IntConstant[val=0](7,36)
> 	Function[isProto=1]	Int(9,0)
> 		Identifier[name:different_arg_names](9,4)
> 		Parameter		Int(9,24)
> 			Identifier[name:a](9,28)
> 		Parameter		Int(9,31)
> 			Identifier[name:b](9,35)
> 	Function[isProto=0]	Int(11,0)
> 		Identifier[name:different_arg_names](11,4)
> 		Parameter		Int(11,24)
> 			Identifier[name:b](11,28)
> 		Parameter		Int(11,31)
> 			Identifier[name:a](11,35)
> 		Scope(11,38)
> 			ReturnStmt(12,2)
> 				Int Expression(12,9)
> 					IntConstant[val=0](12,9)
